### DISCUSSION TITLE
How to overwrite checkpoints with ModelCheckpoint

### DISCUSSION THREAD

**Discussion by nsabir2011** [Lightning-AI/pytorch-lightning#15803] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I am currently trying extract features for some images with Resnet18. After that I'm adding a `registered_buffer` to the model. I'm saving all the embeddings for the extracted feature in that buffer after doing some computations. I only do this for 1 epoch and stop the training (although no training is done, only doing feature extraction).

Later, I do infernece and compare the embeddings. Basically, I look for anomaly this way.

Currently, I save the weights like os:
```python
checkpoint = ModelCheckpoint(
        dirpath=os.path.join(result_path, "weights"),
        filename="model",
        monitor=None,
        mode="max",
        auto_insert_metric_name=False,
    )

callbacks.append(checkpoint)
```
This way, a `model.ckpt` is saved and on each training run, a new weight file saved (Ex. `model-v1.ckpt`) if a checkpoint already exists.
However, I don't want to save a new model. I want to overwrite the previous one. How do I do that?

I have tried different combination of args like `save_top_k=0`, `mode="min"`, `every_n_epoch=1` and `save_on_train_epoch_end=True`. But it doesn't work.

How do I achieve what I require?

> PyTorch Lightning Version: 1.70

**Comment by chaitanya100100** [comment]
Any answer on this? This looks like a very fair feature request.
